simulation.cpp

```diff
@@ -265,6 +272,15 @@ std::function<double(std::vector<double>)> get_current_env_function(const simula
     return e.get_current_function();
 }
 
+void perform_environment_change(simulation &s)
+{
+  switch_optimal_function(s);
+
+  if(s.get_input().size() > 1){
+    s.switch_env_indicator();
+    assign_inputs(s);}
```

Comment:
Isn't this called regardless also in `assign_new_inputs()` in `tick()`?

Reply:
Yes, but it still needs to be there for when `perform_environment_change` is called outside of `tick()` (i.e. in tests)